### PR TITLE
UI components for lobby

### LINKED ISSUE #13: Create frontend UI components for lobby
Construct a bunch of custom UI components to be used throughout the frontend of the application. Specifically, the focus is on the minimum amount of components that are needed to get the lobby screen functioning end-to-end, which is the goal for Sprint 1.

-  [x] Button
Screen Shot 2021-04-27 at 3 26 48 PM

-  [x] TextField
-  [x] PlayerList

**Additional notes**

[Figma](https://www.figma.com/file/7sUdrcJ9FfW7r2gY1UUb07/750---Nerds-Oppose-Society?node-id=0%3A1)

### PR DESCRIPTION
**Pull request by dikunong**
Adds Button, TextField and PlayerList (+ Player), which are the custom UI components needed to build the landing page, nickname screen and lobby screen.

Closes #13

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
 import "./App.css";
 
 const browserHistory = createBrowserHistory();
 export const BrowserHistoryContext = createContext(browserHistory);
 
 function App() {
+  const [testText, setTestText] = useState("");
+
+  const clickBtn = () => {
+    console.log("Beep, boop");
```
Comment (RawiriHohepa) [inline frontend/src/App.tsx:18]: You may need to remove these console.log statements for the Github checks to pass

**Review by RawiriHohepa — DISMISSED** [review]
Looks good to me and runs well on my machine, nice job. May need to remove the console.log statements though

File: frontend/src/components/TextField/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import styles from "./style.module.css";
+
+interface TextFieldProps {
+  label: string;
+  textValue: string;
+  size?: "big" | "small";
+  // eslint-disable-next-line no-unused-vars
```
Comment (taitfuller) [inline frontend/src/components/TextField/index.tsx:8]: This issue is caused by ESlint being a dummy & was fixed in a2d2f6b by adding the following to `frontend/.eslintrc.json`:

```
"no-unused-vars": "off",
"@typescript-eslint/no-unused-vars": ["error"],
```

I recommend removing this line by merging those changes from a2d2f6b

File: frontend/src/components/PlayerList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  let showScores: boolean = true;
+
+  if (gameState === "lobby") {
+    showScores = false;
+  }
+
+  // TODO: Retrieve players from backend
+  const me: string = "AlexVerkerk";
+
+  const dummyPlayers: {
```
Comment (taitfuller) [inline frontend/src/components/PlayerList/index.tsx:19]: Did you intend to set this component up to use Context after this PR?

File: frontend/src/components/PlayerList/Player/style.module.css [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  display: grid;
+  font-size: 24px;
+  grid-template-columns: auto 56px 56px;
+  grid-template-rows: 54px 2px;
+  height: 56px;
+  width: 100%;
+}
+
+/*
+ * I'm not sure why, but CSS Modules snake-case selection seems to have broken in here. Grr
```
Comment (taitfuller) [inline frontend/src/components/PlayerList/Player/style.module.css:12]: Sad 😢

File: frontend/src/components/Button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import styles from "./style.module.css";
+
+interface ButtonProps {
+  variant: "primary" | "secondary";
```
Comment (taitfuller) [inline frontend/src/components/Button/index.tsx:5]: Thoughts on making the variant optional & default to "primary"?

**Review by taitfuller — COMMENTED** [review]
Looks sick man! Just got a couple suggestions / questions for ya:

File: frontend/src/components/TextField/index.tsx (hunk 2 above)
Comment (dikunong) [reply frontend/src/components/TextField/index.tsx:8]: Oh yep, totally forgot to remedy this after merging in the changes

File: frontend/src/components/PlayerList/index.tsx (hunk 3 above)
Comment (dikunong) [reply frontend/src/components/PlayerList/index.tsx:19]: Yeah, the idea is that when our context providers are set up it'll be `const players = useContext(...)`. Similarly, `me` could be retrived from localStorage or wherever we keep that.

File: frontend/src/components/Button/index.tsx (hunk 5 above)
Comment (dikunong) [reply frontend/src/components/Button/index.tsx:5]: Clean. Will do

File: frontend/src/App.tsx (hunk 1 above)
Comment (dikunong) [reply frontend/src/App.tsx:18]: For now I'm chucking `// eslint-disable-next-line` to mute the warning. This is obviously ugly but the components are just there for temporary testing and will be removed in my next PR.

File: frontend/src/components/PlayerList/index.tsx (hunk 3 above)
Comment (taitfuller) [reply frontend/src/components/PlayerList/index.tsx:19]: Cool sounds good!

File: frontend/src/App.tsx (hunk 1 above)
Comment (taitfuller) [reply frontend/src/App.tsx:19]: No worries!

**Review by taitfuller — APPROVED** [review]
Nice work man!

**Pull request merged**